daemon/src/model/cfd.rs

```diff
@@ -737,22 +738,24 @@ impl Cfd {
         Ok(self.event(event))
     }
 
-    // TODO: Pass the entire enum
-    pub fn roll_over(self, rollover_result: Result<Dlc>) -> Result<Event> {
+    pub fn roll_over(self, rollover_completed: RolloverCompleted) -> Result<Event> {
         // TODO: Compare that the version that we started the rollover with is the same as the
         // version now. For that to work we should pass the version into the state machine
         // that will handle rollover and the pass it back in here for comparison.
+        // Daniel says: I don't think the version check will help in addition to what we check in
+        // can_rollover. What difference would it make? If we are completed rollover and are
+        // still able to apply it we should, otherwise we might produce inconsistent state with the
+        // other party.
```

Comment:
Looking forward to the day where we have entire conversations incode  comments :rofl:

Reply:
Thomas says: For now I would make it a hard requirement to not allow any concurrency within these commands. If we find this restrictive, we can always relax this later. If the version we started rollover with is not the same as here, all kinds of state changes could have happened and reasoning about this is **not** trivial. Best to fail early here.

Reply:
I will do a version check here.

Reply:
Not sure it's worth it, you will have to pull the version through into the rollover actor and whatnot... But feel free to do it :D

Reply:
I will contradict myself. Let's introduce this version check (or not) in a separate PR. Here's an issue where we can continue to discuss this particular case: https://github.com/itchysats/itchysats/issues/1038.